src/tokens/saga.ts

```diff
@@ -177,6 +178,18 @@ export async function fetchTokenBalancesForAddress(
   )
 }
 
+export async function fetchTokenBalancesForAddressByTokenId(address: string) {
+  const fetchedTokenBalances: FetchedTokenBalance[] = await fetchTokenBalancesForAddress(address)
+  const fetchedBalancesByTokenId = fetchedTokenBalances.reduce(
+    (acc: Record<string, FetchedTokenBalance>, token) => {
+      acc[token.tokenId] = token
+      return acc
+    },
+    {}
```

Comment:
nit: pet peeve of mine, reduce makes my eyes bleed. I think it's nice that's it succint, but it's usually harder to read. And code is read more than written.

Please simple loops 😢

Reply:
🤣 
Yeah, I don't like `reduce` neither. I'll change it.